app/mailers/host_report_mailer.rb

```diff
@@ -0,0 +1,52 @@
+require 'uri'
+
+class HostReportMailer < ApplicationMailer
+  helper ApplicationHelper
+
+  # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
+  def summary(options = {})
+    raise ::Foreman::Exception, N_("Must specify a valid user with email enabled") unless (user = User.find(options[:user]))
+    hosts = User.as user do
+      Host::Managed.authorized_as(user, :view_hosts, Host).order(:last_report).limit(100)
+    end
+
+    @format = options[:format]
+    time = options[:time] || 1.day.ago
+    host_data = HostReport.summarise(time, hosts, @format)
+    total_metrics = load_metrics(host_data)
+    total = 0
+    total_metrics.each_value { |v| total += v }
+
+    @hosts = host_data
+    @timerange = time
+    @out_of_sync = hosts.out_of_sync.sort
+    @disabled = hosts.alerts_disabled.sort
```

Comment:
Don't use these legacy methods, they will be removed in the future versions and they actually return data from the legacy data model not from host reports.

See https://github.com/theforeman/foreman_host_reports/pull/37/files method outofsync_hosts method. I will move this from the helper to host so you can use it directly on host. My PR needs to be merged first then.